src/parquet/file/metadata.h

```diff
@@ -91,6 +91,32 @@ class PARQUET_EXPORT RowGroupMetaData {
 
 class FileMetaDataBuilder;
 
+struct PARQUET_EXPORT FileVersion {
+  /// Application that wrote the file. e.g. "IMPALA"
+  std::string application;
+
+  /// Version of the application that wrote the file, expressed in three parts
+  /// (<major>.<minor>.<patch>). Unspecified parts default to 0, and extra parts are
+  /// ignored. e.g.:
+  /// "1.2.3"    => {1, 2, 3}
+  /// "1.2"      => {1, 2, 0}
+  /// "1.2-cdh5" => {1, 2, 0}
+  struct {
+    int major;
+    int minor;
+    int patch;
+  } version;
+
+  FileVersion() {}
+  explicit FileVersion(const std::string& created_by);
+
+  /// Returns true if version is strictly less than <major>.<minor>.<patch>
+  bool VersionLt(int major, int minor = 0, int patch = 0) const;
```

Comment:
I will make the changes later to suit PARQUET-793.
I made a mistake in my earlier statement. `FileVersion` should be `Version`.
I will make changes on lines of what parquet-mr `SemanticVersion` class does, something as follows:
```
static FileMetaData::Version MR_PARQUET_129_FIXED_VERSION = FileMetaData::Version(1, 2, 9, "parquet-mr");
static FileMetaData::Version CDH_PARQUET_251_FIXED_VERSION = FileMetaData::Version(2, 5, 1, "cdh-5.5");

const FileMetaData::Version& version = file_metadata_->writer_version();
if (version.compare(MR_PARQUET_129_FIXED_VERSION) < 0) {
}
if (version.compare(CDH_PARQUET_255_FIXED_VERSION) < 0) {
}
```

Now `compare` in `SemanticVersion` is generalized as
```
int compare(FileMetadata::Version& o) {
   cmp = compareIntegers(major, o.major);
   if (cmp != 0) { return cmp; }

   cmp = compareIntegers(minor, o.minor);
   if (cmp != 0) { return cmp; }

   cmp = compareIntegers(patch, o.patch);
   if (cmp != 0) { return cmp; }

   cmp = compareString(o.application, application);
    if (cmp != 0) { return cmp; }
}
```